rs/tla_instrumentation/tla_instrumentation/tests/structs.rs

```diff
@@ -0,0 +1,291 @@
+                            ) => start_counter.max(end_counter).clone(),
+                            _ => Nat::from(0_u64),
+                        },
+                    )
+                    .max();
+                let constants = BTreeMap::from([
+                    (
+                        "MAX_COUNTER".to_string(),
+                        max_counter.unwrap_or(Nat::from(0_u64)).to_tla_value(),
+                    ),
+                    (
+                        "My_Method_Process_Ids".to_string(),
+                        BTreeSet::from([PID.to_string()]).to_tla_value(),
+                    ),
+                ]);
+                let outgoing = format!("{}_to_{}", CAN_NAME, "othercan");
+                let outgoing = outgoing.as_str();
+                let incoming = format!("{}_to_{}", "othercan", CAN_NAME);
+                let incoming = incoming.as_str();
+                for pair in trace {
+                    for s in [&mut pair.start, &mut pair.end] {
+                        if !s.0 .0.contains_key(outgoing) {
+                            s.0 .0.insert(
+                                outgoing.to_string(),
+                                Vec::<TlaValue>::new().to_tla_value(),
+                            );
+                        }
+                        if !s.0 .0.contains_key(incoming) {
+                            s.0 .0.insert(
+                                incoming.to_string(),
+                                BTreeSet::<TlaValue>::new().to_tla_value(),
+                            );
+                        }
+                    }
+                }
+                TlaConstantAssignment { constants }
+            },
+        }
+    }
+}
+
+use tla_stuff::{my_f_desc, CAN_NAME, PID, TLA_INSTRUMENTATION_STATE, TLA_TRACES};
+
+struct StructCanister {
+    pub counter: u64,
+}
+
+static mut GLOBAL: StructCanister = StructCanister { counter: 0 };
+
+fn call_maker() {
+    tla_log_request!(
+        "WaitForResponse",
+        Destination::new("othercan"),
+        "Target_Method",
+        2_u64
+    );
+    tla_log_response!(
+        Destination::new("othercan"),
+        TlaValue::Variant {
+            tag: "Ok".to_string(),
+            value: Box::new(3_u64.to_tla_value())
+        }
+    );
+}
+
+impl StructCanister {
+    #[tla_update_method(my_f_desc())]
+    pub async fn my_method(&mut self) {
+        self.counter += 1;
+        let mut my_local: u64 = self.counter;
+        tla_log_locals! {my_local: my_local};
+        call_maker();
+        self.counter += 1;
+        my_local = self.counter;
+        // Note that this would not be necessary (and would be an error) if
+        // we defined my_local in default_end_locals in my_f_desc
+        tla_log_locals! {my_local: my_local};
+    }
+}
+
+fn set_java_path() {
+    let p = std::env::var("PATH").unwrap();
+    let j = std::env::var("JAVABASE").unwrap();
+    std::env::set_var("PATH", format!("{p}:{j}/bin"));
+}
```

Comment:
Can you explain what this does and why it's necessary, and give more meaningful variable names?

Reply:
Added a comment/renamed the vars